### PR TITLE
Add parasites returning to opened eggs

### PR DESCRIPTION
**Pull request by Vermidia**
## About the PR

Using a parasite on an open egg now returns that parasite to the egg (it deletes them). A parasite clicking on an open egg will also return itself to the egg.

Also changed the previous can't infect with dead parasite line to be more inline with the other xeno popups.

## Why / Balance

Implements part of #2092, CM-13 parity

## Technical details

Used AfterInteractUsing and ActivateInWorldEvent (with some logic in the Open method) to check for the parasite component and return the parasite (if it isn't dead) and change the egg back to it's grown state.

There was a double pop-up problem (see video below) but it's fixed now

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/user-attachments/assets/ac6edbf5-d148-46ed-8fe0-9c650a005841

## Breaking changes

no?
**Changelog**

:cl:
- add: Parasites can crawl back into open eggs, and drone castes can also put them back in

### CODE REVIEW

**Comment by Dutch-VanDerLinde** [comment]
We can't infect the egg!

**Comment by deltanedas** [comment]
handle the event after making popup so it doesnt do it twice
and order the egg system to run before infecting

**Comment by Vermidia** [comment]
Problem is they're different events
The parasite is the one saying it can't infect (Due to afterinteracteevent)
But the egg is the one giving the other messages.

I'm not sure how to stop the former

**Comment by Vermidia** [comment]
Nevermind I fixed it by changing the AfterInteractUsingEvent to just InteractUsingEvent so it runs before AfterInteract

File: Content.Shared/_RMC14/Xenonids/Egg/XenoEggSystem.cs [hunk 1]
```diff
@@ -229,13 +230,49 @@ private void OnXenoEggAfterInteract(Entity<XenoEggComponent> egg, ref AfterInter
...
         // TODO RMC14 multiple hive support
-        if (!HasComp<XenoComponent>(args.User) || !HasComp<HandsComponent>(args.User))
+        if (!HasComp<XenoParasiteComponent>(args.User) && (!HasComp<XenoComponent>(args.User) || !HasComp<HandsComponent>(args.User)))
             return;
 
         if (Open(egg, args.User, out _))
             args.Handled = true;
     }
 
+    private void OnXenoEggInteractUsing(Entity<XenoEggComponent> egg, ref InteractUsingEvent args)
```
Comment (DrSmugleaf) [inline Content.Shared/_RMC14/Xenonids/Egg/XenoEggSystem.cs:240]: This will also put them back in and delete them even if they are a player-controlled parasite, 13 doesn't even let you pick those up but for us it might be better to check if there is a player controlling the parasite, and if so to not run this code

File: Content.Shared/_RMC14/Xenonids/Egg/XenoEggSystem.cs [hunk 2]
```diff
@@ -229,13 +230,49 @@ private void OnXenoEggAfterInteract(Entity<XenoEggComponent> egg, ref AfterInter
     private void OnXenoEggActivateInWorld(Entity<XenoEggComponent> egg, ref ActivateInWorldEvent args)
     {
         // TODO RMC14 multiple hive support
-        if (!HasComp<XenoComponent>(args.User) || !HasComp<HandsComponent>(args.User))
+        if (!HasComp<XenoParasiteComponent>(args.User) && (!HasComp<XenoComponent>(args.User) || !HasComp<HandsComponent>(args.User)))
```
Comment (DrSmugleaf) [inline Content.Shared/_RMC14/Xenonids/Egg/XenoEggSystem.cs:233]: Is this change necessary or a leftover from something else?

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: Content.Shared/_RMC14/Xenonids/Egg/XenoEggSystem.cs (hunk 2 above)
Comment (Vermidia) [reply Content.Shared/_RMC14/Xenonids/Egg/XenoEggSystem.cs:233]: Parasites don't have hands so I added it to skip over the early return.

**Comment by KalimbaMachine** [comment]
i have a very specific issue lorewise with calling the parasites "child",
but thats only my opinion.
otherwise its neat.

**Comment by Vermidia** [comment]
It's the CM-13 terminology so I went with it to be consistent

**Review by DrSmugleaf — APPROVED** [review]
Thanks for the PR!
